plaso/lib/line_reader_file.py

```diff
@@ -165,3 +165,39 @@ def tell(self):
       int: current offset into the file-like object.
     """
     return self._current_offset
+
+
+class BinaryDSVReader(object):
+  """Basic reader for delimiter separated text files of unknown encoding.
+
+  This is used for reading data from text files where the content is unknown, or
+  possibly mixed.
+  """
+
+  def __init__(self, binary_line_reader, delimiter):
+    """Initializes the delimited separated values reader.
+
+    Args:
+      binary_line_reader (BinaryLineReader): a binary file reader
+      delimiter (bytes): field delimiter.
+    """
+    super(BinaryDSVReader, self).__init__()
+    self._line_reader = binary_line_reader
+    self._delimiter = delimiter
+
+  def __iter__(self):
+    """Returns separated values.
+
+    Yields:
+      list(bytes): lines of encoded bytes.
+    """
+    line = self._line_reader.readline()
+    while line:
+      fields = line.split(self._delimiter)
+
+      # Strip off the end-of-line character, to match Python 2 CSV
```

Comment:
character => indicator

seen the end-of-line can consist of multiple characters such as on Windows

Reply:
Done